addon/mixins/in-viewport.js

```diff
@@ -37,6 +37,8 @@ export default Mixin.create({
     }, this._buildOptions());
 
     setProperties(this, options);
+    this._triggerDidScrollDirectionHandler = this._triggerDidScrollDirectionHandler.bind(this);
+    this._setViewportEnteredHandler = this._setViewportEnteredHandler.bind(this);
```

Comment:
This somewhat spreads out the logic wrt to setViewportEntered.  What do ou think?

Reply:
do you have any better idea how to remove handlers?